section_faiss_vector_index.go

```diff
@@ -619,13 +557,19 @@ func (vo *vectorIndexOpaque) process(field index.VectorField, fieldID uint16, do
 
 // todo: better hash function?
 // keep the perf aspects in mind with respect to the hash function.
-// random seed based hash golang.
+// Uses a time based seed to avoid 2 identical vectors in different
+// segments having the same hash -> could be an issue when merging those
+// segments
 func hashCode(a []float32) int64 {
-	var rv int64
+	var rv, sum int64
 	for _, v := range a {
-		rv = int64(math.Float32bits(v)) + (31 * rv)
+		sum += int64(math.Float32bits(v))
```

Comment:
could you please explain why the operation (31 * rv) is dropped? like from what i remember this formula i borrowed from java sdk, and from what i understood this factor was something that helped in generating a better hash than sum of values in an array.

Reply:
yes, that's the reason - the main differentiator is now the combination of vector sum + random int. 
Even if 2 different vectors sum to the same number, the random int in the other 32 bits should be the differentiator.